randomActivityGen.py

```diff
@@ -225,36 +236,78 @@ def verify_stats(stats: ET.ElementTree):
+
+    # Setup file logger, use given or default filename, and overwrite logs on each run
+    log_file_handler = logging.FileHandler(filename=args["--log-file"], mode="w")
+    # Use more verbose format for logfile
+    log_file_handler.setFormatter(logging.Formatter("%(asctime)s %(levelname)-8s %(message)s"))
+    log_stream_handler.setFormatter(stream_formatter)
+
+    # Parse log-level
+    if args["--quiet"]:
+        log_level = logging.ERROR
+    elif args["--verbose"]:
+        log_level = logging.DEBUG
+    else:
+        log_level = getattr(logging, str(args["--log-level"]).upper())
+
+    # Set log-levels and add handlers
+    log_file_handler.setLevel(log_level)
+    logger.addHandler(log_stream_handler)
+    logger.setLevel(log_level)
+
+    # FIXME: logfile should always print in DEBUG, this seems like a larger hurdle:
+    # https://stackoverflow.com/questions/25187083/python-logging-to-multiple-handlers-at-different-log-levels
+    log_file_handler.setLevel(logging.DEBUG)
+    logger.addHandler(log_file_handler)
+
+    # Parse random and seed arguments
     if not args["--random"]:
         random.seed(args["--seed"])
     # The 'noise' lib has good resolution until above 10 mil, but a SIGSEGV is had on values above [-100000, 100000]
     import perlin
     perlin.POPULATION_BASE = random.randint(0, 65_536)
     perlin.INDUSTRY_BASE = random.randint(0, 65_536)
-    while perlin.POPULATION_BASE == perlin.INDUSTRY_BASE: perlin.INDUSTRY_BASE = random.randint(0, 65_536)
+    while perlin.POPULATION_BASE == perlin.INDUSTRY_BASE:
+        perlin.INDUSTRY_BASE = random.randint(0, 65_536)
+    logging.debug(f"Using POPULATION_BASE: {perlin.POPULATION_BASE}, INDUSTRY_BASE: {perlin.INDUSTRY_BASE}")
 
     # Read in SUMO network
+    logging.info(f"Reading network from: {args['--net-file']}")
     net = sumolib.net.readNet(args["--net-file"])
 
     # Parse statistics configuration
+    logging.info(f"Parsing stat file: {args['--stat-file']}")
     stats = ET.parse(args["--stat-file"])
     verify_stats(stats)
 
     # Scale and octave seems like sane values for the moment
+    logging.info("Writing Perlin noise to population and industry")
     apply_network_noise(net, stats)
 
+    logging.info(f"Setting up {int(args['--gates.count'])} city gates ")
     setup_city_gates(net, stats, int(args["--gates.count"]))
 
     if args["--schools.count"] == "auto":
+        logging.info("Setting up schools automatically")
         setup_schools(net, stats, None)
     else:
+        logging.info(f"Setting up {int(args['--schools.count'])} schools")
         setup_schools(net, stats, int(args["--schools.count"]))
 
     # Write statistics back
+    logging.info(f"Writing statistics file to {args['--output-file']}")
     stats.write(args["--output-file"])
 
     if args["--display"]:
-        display_network(net, stats, 500, 500)
+        x_size, y_size = 500, 500
+        logging.info(f"Displaying network as image sized: {x_size} x {y_size}")
```

Comment:
The produced image is not guaranteed to be `x_size` x `y_size`. The arguments are *max* width and height as the city might not be quadratic.

Reply:
Yep, I forgot that. Fixed in 39bfdb7